qp_deblur/deblur.py

```
@@ -73,20 +74,81 @@ def generate_deblur_workflow_commands(preprocessed_fp, out_dir, parameters):
+    param_alignment = ''
+    if reference_alignment is not None:
+        param_alignment = ' -a %s ' % abspath(reference_alignment)
+    # SEPP writes output into the current working directory (cwd), therefore
+    # we here first need to store the cwd, then move into the output directory,
+    # perform SEPP and move back to the stored cwd for a clean state
+    curr_pwd = environ['PWD']
+    std_out, std_err, return_value = system_call(
+        'cd %s && run-sepp.sh %s %s -x %i %s %s; cd %s' %
+        (out_dir, file_input, run_name, threads,
+         param_phylogeny, param_alignment, curr_pwd))
+
+    # parse placements from SEPP results
+    file_placements = '%s/%s_placement.json' % (out_dir, run_name)
+    if exists(file_placements):
+        with open(file_placements, 'r') as fh_placements:
+            plcmnts = json.loads(fh_placements.read())
+            return {p['nm'][0][0]: p['p'] for p in plcmnts['placements']}
+    else:
+        # due to the wrapper style of run-sepp.sh the actual exit code is never
+        # returned and we have no way of finding out which sub-command failed
+        # Therefore, we can only assume that something went wrong by not
+        # observing the expected output file.
+        # If the main SEPP program fails, it reports some information in two
+        # files, whoes content we can read and report
+        file_stderr = '%s/sepp-%s-err.log' % (out_dir, run_name)
+        if exists(file_stderr):
+            with open(file_stderr, 'r') as fh_stderr:
+                std_err = fh_stderr.readlines()
+        file_stdout = '%s/sepp-%s-out.log' % (out_dir, run_name)
+        if exists(file_stdout):
+            with open(file_stdout, 'r') as fh_stdout:
+                std_out = fh_stdout.readlines()
+        error_msg = ("Error running run-sepp.sh:\nStd out: %s\nStd err: %s"
+                     % (std_out, std_err))
+        return False, None, error_msg
```

Actually - I don't think this is going to work. It doesn't conform with the function signature. It should return a dictionary, not 3 values. I think it should raise an error and the caller function catching and doing this return.

---

make sense to me. What type of error should I raise here?

---

not sure - ValueError? just make sure to modify the caller to catch the error.

---

how about a ChildProcessError ?

---

Sure